internal/infrastructure/mailer/smtp.go

```diff
@@ -0,0 +1,123 @@
+package mailer
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/smtp"
+	"net/textproto"
+	"strings"
+
+	"github.com/reearth/reearth-backend/internal/usecase/gateway"
+)
+
+type smtpMailer struct {
+	host     string
+	port     string
+	username string
+	password string
+}
+
+type message struct {
+	to           []string
+	subject      string
+	plainContent string
+	htmlContent  string
+	content      *bytes.Buffer
+}
+
+func (m *message) encodeContent() error {
```

Comment:
```suggestion
func (m *message) encodeContent() (string, error) {
```

Avoid functions with side effects as possible except methods of entities. It makes code hard to understand to reduce bugs and human errors (for example, you might forget to call a function you need).

This function essentially can be turned into a pure function because it exec a simple operation that inputs and then outputs some values.

This change will eliminate the need for content field in message struct.